### PR TITLE
Fix: handle additional metadata

### PR DESCRIPTION
**Pull request by ma-zal**
### Change Log

- Fixed: References `connection`, `altConnection`, `webhook` are correctly deployed to Make
  during remote component creation and during "Deploy to Make".
- Fixed: New Remote Instant Trigger Module creation works correctly now.

### CODE REVIEW

File: src/local-development/create-remote-component.ts [hunk 1]
```diff
@@ -1,49 +1,142 @@
 import { AxiosRequestConfig } from 'axios';
-import { AppComponentMetadata, LocalAppOriginWithSecret } from './types/makecomapp.types';
```
Comment (JakeHamix) [inline src/local-development/create-remote-component.ts:2]: When importing just types, I'd prefer sticking to the import type syntax to make it more apparent.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html

File: src/local-development/create-remote-component.ts [hunk 2]
```diff
@@ -1,49 +1,142 @@
...
+			url: componentCreationUrl,
+			method: 'POST',
+			// Add all editable component metadata
+			data: getComponentRemoteMetadataToDeploy('module', opt.componentMetadata, opt.makecomappJson, opt.origin),
+		};
+
+		// Add metadata, which are not covered by `getComponentRemoteMetadataToDeploy`,
+		//   because there are persistent/not-editable after component creation.
+
+		if (opt.componentType === 'module') {
```
Comment (JakeHamix) [inline src/local-development/create-remote-component.ts:56]: Switch/case? 🙏

File: src/local-development/deploy-metadata.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
+import { AxiosRequestConfig } from 'axios';
+import { AppComponentMetadata, LocalAppOriginWithSecret, MakecomappJson } from './types/makecomapp.types';
+import { getComponentApiUrl } from './helpers/api-url';
+import { ComponentIdMappingHelper } from './helpers/component-id-mapping-helper';
+import { log } from '../output-channel';
+import { AppComponentType } from '../types/app-component-type.types';
```
Comment (JakeHamix) [inline src/local-development/deploy-metadata.ts:6]: import type

**Review by JakeHamix — APPROVED** [review]
The logs and comments are 🔥

File: src/local-development/create-remote-component.ts (hunk 2 above)
Comment (ma-zal) [reply src/local-development/create-remote-component.ts]: Done

File: src/local-development/create-remote-component.ts (hunk 1 above)
Comment (ma-zal) [reply src/local-development/create-remote-component.ts:2]: Done

**Pull request merged**